pkg/controller/controller.go

```diff
@@ -583,6 +588,10 @@ func (h *Handler) syncDiskMakerDaemonset(cr *localv1.LocalVolume, forceRollout b
 	ds := h.generateDiskMakerDaemonSet(cr)
 	dsName := ds.Name
 	generation := getExpectedGeneration(cr, ds)
+
+	// See if we need to roll out the pods based on the DaemonSet hash
+	forceRollout = h.checkDaemonSetHash(ds)
```

Comment:
This variable here overrides `forceRollout` parameter supplied to the function. This will create problems in cases where daemonset spec does not have any change but `LocalVolume` object has changed.

Reply:
This has been updated to be its own `bool`. If the new `daemonSetUpdated` is true, then `forceRollout` will be updated as well; otherwise, `forceRollout` keeps its original value.

Reply:
This does mean that we did not had had adequate tests to cover these scenarios. Can we make sure that this is covered?

Reply:
I moved this logic into `checkDaemonSetHash()`, and updated `TestCheckDaemonSetHash` to examine a combination of `forceRollout` and updated daemon sets. Let me know your thoughts on the current approach.